### PR TITLE
Cpf 4796/role approval request migration FOAM

### PR DESCRIPTION
**Pull request by tharmaman**
Basically removed RoleApprovalRequest and merged the properties into ApprovalRequest

More importantly, createdBy is used instead of initiatingUser now.

### CODE REVIEW

File: src/foam/nanos/approval/ApprovalRequest.js [hunk 1]
```diff
@@ -78,31 +139,40 @@
...
       },
-      visibility: function(approver) {
-        return approver ?
-          foam.u2.DisplayMode.RO :
-          foam.u2.DisplayMode.HIDDEN;
+      tableCellFormatter: function(approver, data) {
+        let self = this;
+        // If request is REQUESTED, show as Pending
+        // Otherwise, show approver's name
+        if ( data.status === foam.nanos.approval.ApprovalStatus.REQUESTED ) {
```
Comment (jlhughes) [inline src/foam/nanos/approval/ApprovalRequest.js]: previous logic showed the approvers name if they were logged in. 
Example - as a user that can others approvals - payment-ops (or admin) one cannot tell which approval is for them.

File: src/foam/nanos/approval/ApprovalRequest.js [hunk 2]
```diff
@@ -78,31 +139,40 @@
...
+          this.add(data.REQUESTED);
+        } else {
+          this.__subSubContext__.userDAO.find(approver).then(user => {
+            self.add(user ? user.toSummary() : `User #${approver}`);
+          });
+        }
       },
+      visibility: function(status, approver) {
+        if ( status === foam.nanos.approval.ApprovalStatus.REQUESTED || ! approver ) {
+          return foam.u2.DisplayMode.HIDDEN;
```
Comment (jlhughes) [inline src/foam/nanos/approval/ApprovalRequest.js]: This suggests REQUESTED are HIDDEN?

File: src/foam/nanos/approval/ApprovalRequest.js (hunk 1 above)
Comment (tharmaman) [reply src/foam/nanos/approval/ApprovalRequest.js]: Yeah we can revert back to the previous logic I think this was the liquid specific code that had to be added for the demo

File: src/foam/nanos/approval/ApprovalRequest.js (hunk 2 above)
Comment (tharmaman) [reply src/foam/nanos/approval/ApprovalRequest.js]: Again was liquid specific code that was added feel free to remove

File: src/foam/nanos/approval/ApprovalRequest.js (hunk 1 above)
Comment (tharmaman) [reply src/foam/nanos/approval/ApprovalRequest.js]: Feel free to remove

**Review by jlhughes — APPROVED** [review]

**Review by nanoscott — APPROVED** [review]

**Pull request merged**